### DISCUSSION TITLE
Doubt about MachineConfig interface

### DISCUSSION THREAD

**Discussion by vcoopman** [statelyai/xstate#2639] [Q&A] [answered]
```
export interface MachineConfig<
  TContext,
  TStateSchema extends StateSchema,
  TEvent extends EventObject
> extends StateNodeConfig<TContext, TStateSchema, TEvent> {
  /**
   * The initial context (extended state)
   */
  context?: TContext | (() => TContext);
  /**
   * The machine's own version.
   */
  version?: string;
  schema?: MachineSchema<TContext, TEvent>;
}
```

What does " context?: TContext | **(() => TContext**); " means? ( right side of the OR ) 
And why it is there?

**Comment by ChrisShank** [comment, marked as answer]
It allows you to lazily initialize the `context` of a machine. This is necessary when you want to spawn actors when the machine is first initialized! 

https://github.com/statelyai/xstate/blob/main/packages/core/CHANGELOG.md#4220

**Reply by vcoopman** [reply]
thanks!!
